azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/DataDisk.java

```diff
@@ -65,138 +63,251 @@
      */
     DiskCreateOptionTypes createOption();
 
-    /**************************************************************
-     * Fluent interfaces for builder pattern
-     **************************************************************/
+    // fluent (setters)
 
     /**
-     * Container interface for all the definitions.
-     *
-     * @param <ParentT> the return type of the final {@link DefinitionAttachable#attach()}
+     * Grouping of data disk definition stages applicable as part of a virtual machine creation.
      */
-    interface Definitions<ParentT> extends
-            DataDisk.DefinitionAttachNewDataDisk<ParentT>,
-            DataDisk.DefinitionAttachExistingDataDisk<ParentT>,
-            DataDisk.DefinitionWithStoreAt<ParentT>,
-            DataDisk.DefinitionWithOptionalConfiguration<ParentT>,
-            DataDisk.DefinitionAttachable<ParentT> {
-    }
+    interface DefinitionStages {
+        /**
+         * The first stage of a  data disk definition.
+         * @param <ParentT> the return type of the final {@link WithAttach#attach()}
+         */
+        interface Blank<ParentT>
+                extends DefinitionWithDataDisk<ParentT> {
+        }
 
-    /**
-     * The first stage of new data disk configuration.
-     *
-     * @param <ParentT> the return type of the final {@link DefinitionAttachable#attach()}
-     */
-    interface DefinitionAttachNewDataDisk<ParentT> {
         /**
-         * Specifies the initial disk size in GB for new blank data disk.
+         * The stage allowing to choose configuring new or existing data disk.
          *
-         * @param sizeInGB the disk size in GB
-         * @return the stage representing optional additional settings for the attachable data disk
+         * @param <ParentT> the return type of the final {@link WithAttach#attach()}
          */
-        DefinitionWithStoreAt<ParentT> withSizeInGB(Integer sizeInGB);
-    }
+        interface DefinitionWithDataDisk<ParentT>
```

Comment:
Any chance this could also follow the pattern of not repeating the word Definition when it's already inside an interface called DefinitionStages?  In other words: DefinitionStages.WithDataDisk as opposed to DefinitionStages.DefinitionWithDataDisk?
(the NSG security rule interfaces are organized that way as well)